common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1601,3 +1602,72 @@ def can_toggle_course_pacing(self):
           bool: False if the course has already started, True otherwise.
         """
         return datetime.now(UTC()) <= self.start
+
+
+class CourseSummaryDescriptor(object):
```

Comment:
I wouldn't use the term `Descriptor` here. That has a specific meaning for XModules, and I don't think we want to tie this code to that (even just by implication).